testlib/helper.go

```diff
@@ -117,6 +117,7 @@ func (h *MainHelper) setupResources() {
 
 func (h *MainHelper) Close() error {
 	if h.Settings != nil {
+		h.SQLSupplier.Close()
```

Comment:
Should we be testing this in a block on `h.SQLSupplier != nil` and not `h.Settings != nil`?

Reply:
Is not strictly necessary, but we can do that, the Settings attribute initialization is binded to the SQLSupplier attribute initialization, but being purist we should check both here.